src/Bicep.Local.Extension/Rpc/GrpcChannelHelper.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Immutable;
+using System.Net;
+using System.Net.Sockets;
+using Grpc.Core;
+using Grpc.Net.Client;
+
+namespace Bicep.Local.Extension.Rpc;
+
+public static class GrpcChannelHelper
+{
+    public class UnixDomainSocketsConnectionFactory
+    {
+        private readonly EndPoint endPoint;
+
+        public UnixDomainSocketsConnectionFactory(EndPoint endPoint)
+        {
+            this.endPoint = endPoint;
+        }
+
+        public async ValueTask<Stream> ConnectAsync(SocketsHttpConnectionContext _,
+            CancellationToken cancellationToken = default)
+        {
+            var socket = new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified);
+
+            try
+            {
+                await socket.ConnectAsync(this.endPoint, cancellationToken).ConfigureAwait(false);
+                return new NetworkStream(socket, true);
+            }
+            catch
+            {
+                socket.Dispose();
+                throw;
+            }
+        }
+    }
+
+    public static GrpcChannel CreateChannel(string socketPath)
+    {
+        var udsEndPoint = new UnixDomainSocketEndPoint(socketPath);
+        var connectionFactory = new UnixDomainSocketsConnectionFactory(udsEndPoint);
+        var socketsHttpHandler = new SocketsHttpHandler
+        {
+            ConnectCallback = connectionFactory.ConnectAsync
+        };
+
+        return GrpcChannel.ForAddress("http://localhost", new GrpcChannelOptions
```

Comment:
Is the URL actually used since the channel is connecting over a domain socket? If it isn't, I might make it something unresolvable like `never://this-should-not-be-used.wtf` so that the whole thing fails fast and the FQDN jumps out in log messages in case this ever does not work as expected.

Reply:
This is taken from the official docs: https://learn.microsoft.com/en-us/aspnet/core/grpc/interprocess-uds?view=aspnetcore-8.0. I'll add a comment here to explain this, as it's not very clear from the code that it's unused. I'd rather stick with localhost than make up a domain which may confuse people further - especially if I don't think we'd hit this in production.

Reply:
I'd be slightly more concerned about the failure mode -- `http://localhost` is just as likely to connect to the wrong thing as it is to fail to connect, which I would think would be more confusing for users (e.g., it'd be reported as something exotic like a GRPC protocol error or validation exception).

This is a pretty far-fetched scenario, though, so feel free to ignore.